AK/IPv6Address.h

```diff
@@ -256,13 +256,101 @@ class [[gnu::packed]] IPv6Address {
         return IPv6Address({ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 });
     }
 
+    [[nodiscard]] constexpr bool is_loopback() const
+    {
+        return *this == loopback();
+    }
+
+    [[nodiscard]] constexpr bool is_in_subnet(IPv6Address subnet, u16 network_size) const
+    {
+        VERIFY(network_size <= 128);
+        return this->network(network_size) == subnet;
+    }
+
+    [[nodiscard]] constexpr IPv6Address network(u16 network_size) const
+    {
+        VERIFY(network_size <= 128);
+        u64 const low_word_size = min(network_size, 64);
+        u64 const high_word_size = clamp(static_cast<i16>(network_size) - 64, 0, 64);
+        // FIXME: somehow avoid branching for special case *without* introducing UB?
+        // Bitmagic works as follows:
+        //  - Invert the network size since we can more easily create many mask bits towards the "high" side of the address (i.e. low bits)
+        //  - For instance, size = 16 => mask off top 48 bits, so first create a mask with 48 1 bits at the bottom.
+        //  - Use inverted size to create corresponding bitmask ((1 << size) - 1). This bitmask is exactly inverse to what we want.
+        //    In the example, inverting gives us all 1 bits in the top 16 positions.
+        u64 const low_word_mask = low_word_size == 0 ? 0uLL : ~((1uLL << (64 - low_word_size)) - 1uLL);
+        u64 const high_word_mask = high_word_size == 0 ? 0uLL : ~((1uLL << (64 - high_word_size)) - 1uLL);
+        u64 const masked_low_word = low_word_mask & low_word();
+        u64 const masked_high_word = high_word_mask & high_word();
```

Comment:
This looks like something that can be simplified by using `u128`.

Reply:
I’m not sure if our u128 is capable enough for this? Will check. (In Rust I would definitely agree.)

Reply:
```cpp

        u128 const word_mask = network_size == 0 ? 0 : ~((u128 { 1uLL } << (128 - network_size)) - u128 { 1uLL });
        u128 masked_word = word_mask & u128 { m_data };
        IPv6Address result;
        TypedTransfer<u8>::move(result.m_data, masked_word.bytes().data(), 16);
        return result;
```
No, this doesn’t work and I don’t know how to fix it.

Reply:
(problem is that the cast to u128 seems to fail for weird reasons, and i can’t figure out why.)